components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/ServerApplicationManagementService.java

```diff
@@ -849,6 +851,16 @@ public String createApplication(ApplicationModel applicationModel, String templa
     public void patchApplication(String applicationId, ApplicationPatchModel applicationPatchModel) {
 
         ServiceProvider appToUpdate = cloneApplication(applicationId);
+        AssociatedRolesConfig.AllowedAudienceEnum allowedAudience;
+        AssociatedRolesConfig associatedRoles = applicationPatchModel.getAssociatedRoles();
+        if (associatedRoles != null) {
+            allowedAudience = associatedRoles.getAllowedAudience();
+            List<Role> listRole = associatedRoles.getRoles();
+            if (AssociatedRolesConfig.AllowedAudienceEnum.ORGANIZATION.equals(allowedAudience)
+                    && !listRole.isEmpty()) {
+                throw buildClientError(ErrorMessage.INVALID_ROLE_ASSOCIATION_FOR_ORGANIZATION_AUDIENCE);
+            }
+        }
```

Comment:
Can't we make `allowedAudience` is the only editable param of AssociatedRolesConfig object?
associatedRoles will be only a readonly attribute
for applicationAudience apps -> return the created App audienced roles, association will be built when creating the app audienced roles
for organizationAudiece apps -> return nothing

Reply:
For organizationAudiece apps, we need to return all the organization roles available in the organization regardless of the audience.